sdcm/nemesis.py

```diff
@@ -2974,52 +2918,61 @@ def wrapper(*args, **kwargs):  # pylint: disable=too-many-statements
+                log_info.update({'error': details, 'full_traceback': traceback.format_exc()})
+                status = False
 
-        except Exception as details:  # pylint: disable=broad-except
-            details = str(details)
-            args[0].error_list.append(details)
-            args[0].log.error('Unhandled exception in method %s', method, exc_info=True)
-            log_info.update({'error': details, 'full_traceback': traceback.format_exc()})
-            status = False
-        finally:
-            end_time = time.time()
-            time_elapsed = int(end_time - start_time)
-            log_info.update({
-                'end': int(end_time),
-                'duration': time_elapsed,
-            })
-            args[0].duration_list.append(time_elapsed)
-            args[0].operation_log.append(log_info)
-            args[0].log.debug('%s duration -> %s s', args[0].current_disruption, time_elapsed)
-
-            if class_name.find('Chaos') < 0:
-                args[0].metrics_srv.event_stop(class_name)
-            disrupt = args[0].get_disrupt_name()
-            del log_info['operation']
-
-            try:  # So that the nemesis thread won't stop due to elasticsearch failure
-                args[0].update_stats(disrupt, status, log_info)
-            except ElasticSearchConnectionTimeout as err:
-                args[0].log.warning(f"Connection timed out when attempting to update elasticsearch statistics:\n"
-                                    f"{err}")
-            except Exception as err:
-                args[0].log.warning(f"Unexpected error when attempting to update elasticsearch statistics:\n"
-                                    f"{err}")
-            DisruptionEvent(type=disrupt, status=status, **log_info).publish()
-            args[0].cluster.check_cluster_health()
-            num_nodes_after = len(args[0].cluster.nodes)
-            if num_nodes_before != num_nodes_after:
-                args[0].log.error('num nodes before %s and nodes after %s does not match' %
-                                  (num_nodes_before, num_nodes_after))
-            # TODO: Temporary print. Will be removed later
-            data_validation_prints(args=args)
+            finally:
+                end_time = time.time()
+                time_elapsed = int(end_time - start_time)
+                log_info.update({
+                    'end': int(end_time),
+                    'duration': time_elapsed,
+                })
+                args[0].duration_list.append(time_elapsed)
+                args[0].operation_log.append(log_info)
+                args[0].log.debug('%s duration -> %s s', args[0].current_disruption, time_elapsed)
+
+                if class_name.find('Chaos') < 0:
+                    args[0].metrics_srv.event_stop(class_name)
+                disrupt = args[0].get_disrupt_name()
+                del log_info['operation']
```

Comment:
from what I see it is not used at all, let's remove it

Reply:
@dkropachev 
As I see from the code, operation data is deleted from `log_info` before Elasticsearch stats update:
`
args[0].update_stats(disrupt, status, log_info)
`

We don't want to save it there. but with operation data we need it for `def report(self)` (saved in the `self.operation_log`)